cpp/include/cugraph/graph.hpp

```diff
@@ -176,6 +176,25 @@ class graph_t<vertex_t, edge_t, weight_t, store_transposed, multi_gpu, std::enab
       false);
   }
 
+  // generic in-place sorter on CSR structure;
+  // this is specifically targetting
+  // segmented-sort by weigths; but
+  // must be generic enough to support future
+  // types of sorting;
+  // Notes:
+  // (1.) criterion is mutable (non-const)
+  //      to allow for sorter obejcts for which
+  //      the sorting operation fills additional structures
+  //      (later to be retrieved; e.g., for debugging);
+  // (2.) sorting object is responsible for updating "in-place"
+  //      any of the (offsets, indices, weights) arrays;
+  //
+  template <typename in_place_sorter_t>
+  void sort(in_place_sorter_t& criterion)
```

Comment:
Is this function still necessary?

Reply:
No, but since it's there, why remove it? It's a piece of functionality that might turn out to be necessary. At least useful. I am testing it.

Reply:
So, sorting (based on specifics) will affect many other algorithms. For example, we may need to sort neighbors of each vertex to facilitate neighbor intersection (necessary for triangle counting and so on).

So, if we allow sorting of internal edge data (with the functor that allows sorting in many different ways), we should go for lengthy discussion about how this affects other algorithms.

Unless we want to go for that discussion in this PR, better delete this at this moment.

Reply:
I agree with both of you.  this might be a useful feature later on, so preserving the code has benefit.  On the other hand, we do not want dead or unused code floating around in the code base.  

simply commenting it out for now, and adding a FIXME tag, is a good middle ground for now so that the PR can get merged.  We can then have a sorting sync and then either delete or improve the code.

Reply:
Done. Commented out.